src/includes/models/class-alma-wc-model-payment.php

```diff
@@ -81,6 +81,7 @@ public static function get_payment_payload_from_order( $order_id, $fee_plan_defi
 		if ( Alma_WC_Payment_Upon_Trigger::does_payment_upon_trigger_apply_for_this_fee_plan( $fee_plan_definition ) ) {
 			$data['payment']['deferred']             = 'trigger';
 			$data['payment']['deferred_description'] = Alma_WC_Payment_Upon_Trigger::get_display_text();
+			update_post_meta( $order_id, 'payment_upon_trigger_enabled', true );
```

Comment:
@gilles66 please :
1. register this `post_meta` when payment is created
2.  link it to the payment object, not the order
3. prefix the `meta_name` (`payment_upon_trigger_enabled`) with `alma_`

Reply:
Point 3 is done but I don't understand your demand for points 1 and 2.
In my opinion there is no payment object.

Reply:
finally, as we shown yesterday, you have to `update_post_meta` on payment validation (after transaction_id was stored in order)

Reply:
Payment validation time seems too late to determine if the order has to be paid uppon trigger because this doesn't depend on the amout or the order, but depends on the fee plan selected and during payment validation this information is unknown. I don't know how to do what you want.

Reply:
ok done.